# Day 11 – NumPy Student Marks

## VEDA Technology AI & ML Internship

### Task Objective
Create a NumPy array containing student marks and calculate:
- Mean
- Median
- Maximum
- Minimum
- Standard Deviation

In [1]:
import numpy as np

In [2]:
marks = np.array([78, 85, 92, 67, 74, 88, 95, 81, 69, 90])

print("Student Marks:")
print(marks)

Student Marks:
[78 85 92 67 74 88 95 81 69 90]


In [3]:
mean_marks = np.mean(marks)

print("Mean:", mean_marks)

Mean: 81.9


In [4]:
median_marks = np.median(marks)

print("Median:", median_marks)

Median: 83.0


In [5]:
maximum_marks = np.max(marks)

print("Maximum:", maximum_marks)

Maximum: 95


In [6]:
minimum_marks = np.min(marks)

print("Minimum:", minimum_marks)

Minimum: 67


In [7]:
standard_deviation = np.std(marks)

print("Standard Deviation:", standard_deviation)

Standard Deviation: 9.23525852372309


In [8]:
print("===== Student Marks Statistics =====")
print("Number of Students:", len(marks))
print("Mean:", mean_marks)
print("Median:", median_marks)
print("Maximum:", maximum_marks)
print("Minimum:", minimum_marks)
print("Standard Deviation:", standard_deviation)

===== Student Marks Statistics =====
Number of Students: 10
Mean: 81.9
Median: 83.0
Maximum: 95
Minimum: 67
Standard Deviation: 9.23525852372309


## Explanation

The NumPy array contains the marks of 10 students.

- **Mean** represents the average marks of all students.
- **Median** represents the middle value of the dataset.
- **Maximum** represents the highest mark obtained.
- **Minimum** represents the lowest mark obtained.
- **Standard deviation** represents how much the marks vary from the mean.

The mean mark is 81.90 and the median is 83.00. The highest mark is 95 and the lowest mark is 67. The standard deviation is approximately 9.04, showing the spread of the marks around the mean.